### PR TITLE
Pretrain and fine-tune a model

### PR DESCRIPTION
**Pull request by zredlined**
Load an existing model, vocab, and tokenizer and fine-tune on new data. Also added a configuration option to evaluate the model on a test set after training.

### CODE REVIEW

**Review by johntmyers — CHANGES_REQUESTED** [review]

**Comment by amysteier** [comment]
How hard to test on the hold out set after each epoch?  Ideally we want to stop training when loss stops going down on this set versus the training set.
